ignite/engine/__init__.py

```diff
@@ -384,27 +420,65 @@ def create_supervised_trainer(
 
         - Added ``amp_mode`` argument for automatic mixed precision.
         - Added ``scaler`` argument for gradient scaling.
+
+    .. versionchanged:: 0.5.0
+        Added Gradient Accumulation argument for all supervised training methods.
     """
 
     device_type = device.type if isinstance(device, torch.device) else device
     on_tpu = "xla" in device_type if device_type is not None else False
     mode, _scaler = _check_arg(on_tpu, amp_mode, scaler)
 
+    if gradient_accumulation_steps == 0:
```

Comment:
This exception should catch if `gradient_accumulation_steps <= 0`.
IMO it should be added in every functions since they should be used standalone.